src/oss/contributing/documentation.mdx

```diff
@@ -0,0 +1,358 @@
+        :::
+    </Step>
+
+    <Step title="Development environment">
+        Set up your development environment following our [IDE setup guide](#ide-setup)
+    </Step>
+
+    <Step title="Understanding the build system">
+        Familiarize yourself with our documentation [build pipeline](#build-system)
+    </Step>
+
+    <Step title="Mintlify knowledge">
+        Review the [Mintlify documentation](https://mintlify.com/docs) for component usage and best practices
+    </Step>
+</Steps>
+
+### IDE setup
+
+This repository includes configuration files to ensure consistent formatting across different editors:
+
+<Tabs>
+    <Tab title="VSCode">
+        Settings in `.vscode/settings.json` are automatically applied when you open the project. No additional setup required.
+
+        **Recommended extensions:**
+        - [EditorConfig for VS Code](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)
+        - [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)
+        - [MDX](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx)
+    </Tab>
+
+    <Tab title="Other IDEs">
+        Most modern IDEs support EditorConfig. The `.editorconfig` file automatically configures:
+        - IntelliJ IDEA / PyCharm / WebStorm
+        - Sublime Text / Atom
+        - Vim / Emacs
+
+        Install the EditorConfig plugin if needed and restart your IDE.
+    </Tab>
+</Tabs>
+
+### Build system
+
+<Warning>
+    Only edit files in `src/` - The `build/` directory is automatically generated!
+</Warning>
+
+Our documentation uses a build pipeline that converts source files to Mintlify format:
+
+<Steps>
+    <Step title="Install dependencies">
+        You will need to install [`uv`](https://docs.astral.sh/uv/getting-started/installation/) and [`mint`](https://mintlify.com/docs/installation) if you haven't already.
+
+        ```bash
+        cd docs
+        uv venv
+        source .venv/bin/activate
+        uv sync --all-groups
+        npm i -g mint
+        ```
+    </Step>
+
+    <Step title="Start development server">
+        Run from the repo root:
+
+        ```bash
+        docs dev
+        ```
+
+        This starts a development server with hot reload at http://localhost:3000 (unless overridden).
+    </Step>
+
+    <Step title="Make your changes">
+        Edit files in `src/` and see changes reflected immediately in your browser.
+    </Step>
+</Steps>
+
+## Documentation types
+
+### Conceptual guides
+
+Conceptual guides fall under the "Explanation" quadrant of Diataxis. They cover LangChain terms and core concepts abstractly, providing deeper understanding.
```

Comment:
These sections are great! Each one should link out to a couple good examples of each